server.h

```diff
@@ -487,14 +520,191 @@ class CTFOServer final {
         }
       }
     });
+
+    // TODO(dkorolev): Avoid this shameless copy-pasting.
+    const auto comments_handler = [this](Request r) {
+      const UID uid = StringToUID(r.url.query["uid"]);
+      const std::string token = r.url.query["token"];
+      const CID cid = StringToCID(r.url.query["cid"]);
+      if (uid == UID::INVALID_USER) {
+        DebugPrint(Printf("[/ctfo/comment] Wrong UID. Requested URL = '%s'", r.url.ComposeURL().c_str()));
+        r("NEED VALID UID-TOKEN PAIR\n", HTTPResponseCode.BadRequest);
+      } else if (cid == CID::INVALID_CARD) {
+        DebugPrint(Printf("[/ctfo/comments] Wrong CID. Requested URL = '%s'", r.url.ComposeURL().c_str()));
```

Comment:
I see you created two endpoints for this handler - `/comment` & `/comments` - I suggest to leave one of them in `DebugPrint`'s.